src/OpenLoco/src/Ui/ToolManager.h

```diff
@@ -1,11 +1,121 @@
 #pragma once
 
 #include "Types.hpp"
+#include "Widget.h"
 #include "Window.h"
 #include <cstdint>
 
 namespace OpenLoco::ToolManager
 {
+    using namespace Ui;
+
+    using ToolEventType_t = uint8_t;
+    namespace ToolEventType
+    {
+        constexpr ToolEventType_t onActivate = 0;
+        constexpr ToolEventType_t onCancel = 1;
+        constexpr ToolEventType_t onMouseMove = 2;
+        constexpr ToolEventType_t onMouseDown = 3;
+        constexpr ToolEventType_t onMouseDrag = 4;
+        constexpr ToolEventType_t onMouseDragEnd = 5;
+        constexpr ToolEventType_t onShiftChanged = 6;
+        constexpr ToolEventType_t onControlChanged = 7;
+        constexpr ToolEventType_t onScroll = 8;
+        constexpr ToolEventType_t onScrollShiftModifier = 9;
+        constexpr ToolEventType_t onScrollControlModifier = 10;
+        constexpr ToolEventType_t onScrollAltModifier = 11;
+        constexpr ToolEventType_t count = 12;
+    };
+
+    enum class ToolFlag : uint16_t
+    {
+        none = 0,
+        keepFlag6 = (1U << 0),
+        gridlines = (1U << 1),
+        gridlinesPersistWithWindow = (1U << 2),
+        dragStartsOnMouseDown = (1U << 3),
+        toolPersistsThroughWindowClose = (1U << 4),
```

Comment:
I was thinking of plugins when I added this, but it adds a lot of complexity. Remove this